tests/test_xml_jats.py

```diff
@@ -53,3 +53,136 @@ def test_get_article_id_raises_exception_when_id_not_found():
+     '          <subject>Cancer Biology</subject>'
+     '          <subject>General Economics</subject>'
+     '        </subj-group>'
+     '      </article-categories>'
+     '    </article-meta>'
+     '  </front>'
+     '</article>',
+     ['Cancer Biology']),
+    ('<article>'
+     '  <front>'
+     '    <article-meta>'
+     '      <article-categories>'
+     '        <subj-group subj-group-type="heading">'
+     '          <subject>Research Article</subject>'
+     '        </subj-group>'
+     '        <subj-group>'
+     '          <subject>General Economics</subject>'
+     '          <subject>Cancer Biology</subject>'
+     '        </subj-group>'
+     '      </article-categories>'
+     '    </article-meta>'
+     '  </front>'
+     '</article>',
+     ['General Economics']),
+    ('<article>'
+     '  <front>'
+     '    <article-meta>'
+     '      <article-categories>'
+     '        <subj-group subj-group-type="heading">'
+     '          <subject>Research Article</subject>'
+     '        </subj-group>'
+     '        <subj-group subj-group-type="subjects">'
+     '          <subject>Cancer Biology</subject>'
+     '        </subj-group>'
+     '        <subj-group subj-group-type="subjects">'
+     '          <subject>General Economics</subject>'
+     '        </subj-group>'
+     '      </article-categories>'
+     '    </article-meta>'
+     '  </front>'
+     '</article>',
+     ['Cancer Biology', 'General Economics']),
+    ('<article>'
+     '  <front>'
+     '    <article-meta>'
+     '      <article-categories>'
+     '        <subj-group subj-group-type="heading">'
+     '          <subject>Research Article</subject>'
+     '        </subj-group>'
+     '        <subj-group subj-group-type="subjects">'
+     '          <subject>Cancer Biology</subject>'
+     '        </subj-group>'
+     '        <subj-group>'
+     '          <subject>General Economics</subject>'
+     '        </subj-group>'
+     '      </article-categories>'
+     '    </article-meta>'
+     '  </front>'
+     '</article>',
+     ['Cancer Biology', 'General Economics']),
+    ('<article>'
+     '  <front>'
+     '    <article-meta>'
+     '      <article-categories>'
+     '        <subj-group subj-group-type="heading">'
+     '          <subject>Research Article</subject>'
+     '        </subj-group>'
+     '        <subj-group subj-group-type="subjects">'
+     '          <subject>Cancer Biology</subject>'
+     '        </subj-group>'
+     '        <subj-group>'
+     '          <subject>Data</subject>'
+     '        </subj-group>'
+     '        <subj-group subj-group-type="subjects">'
+     '          <subject>Housing</subject>'
+     '        </subj-group>'
+     '        <subj-group>'
+     '          <subject>General Economics</subject>'
+     '        </subj-group>'
+     '      </article-categories>'
+     '    </article-meta>'
+     '  </front>'
+     '</article>',
+     ['Cancer Biology', 'Data', 'Housing', 'General Economics']),
```

Comment:
@FAtherden-eLife can you validate these tests work as intended?
We would also need a test with nested tags (such as italic?) within the subject name, which you mentioned before.

Reply:
> @FAtherden-eLife can you validate these tests work as intended?

Yup, they look good.

> We would also need a test with nested tags (such as italic?) within the subject name, which you mentioned before.

Yes, good point. I neglected to include that. @GiancarloFusiello, another (more complex) test case is included [here](https://github.com/libero/publisher/issues/256#issuecomment-535876763).

Reply:
@giorgiosironi, @GiancarloFusiello Do we also need to add a test case which covers the logic set out in the Notes section [here](https://docs.google.com/document/d/1PmLD6NEiqpjKTnjW-n1Hho_G-znp5lW3ba8PGmOHnQE/edit?ts=5d8b6ec9#heading=h.9911g2g4u32r)? i.e. two articles with the same content in `subject` except the case is different (e.g. `Ecology` and `ecology`), the first value being used and second taking the value from the first? Or will this be dealt with elsewhere?

Reply:
@FAtherden-eLife I don't think we need a test case for the Notes scenario. Currently implementing the test case add [here](https://github.com/libero/publisher/issues/256) that deals with styling tags.

Reply:
Agreed on elsewhere, that will be covered by the tests of the requests towards `scholarly-content-detail`; separate from extracting the values which is what we do here.

Reply:
@FAtherden-eLife I'll leave it to you to provide feedback on the extents of the tests that should be covered here, no further concerns from me